src/descriptor.c

```diff
@@ -5,12 +5,52 @@
  *
  **********************************************************************/
 
+#include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 
 #include "descriptor.h"
+#include "defines.h"
+#include "dbusutils.h"
+#include "utils.h"
 
-descriptor_t *descriptor_new (const char *uuid, const char *characteristic_path)
+static void descriptor_handle_unregister_device (DBusConnection *connection, void *data);
+static DBusHandlerResult descriptor_handle_dbus_message (DBusConnection *connection, DBusMessage *message, void *data);
+static void descriptor_get_uuid(void *user_data, DBusMessageIter* iter);
+static void descriptor_get_characteristic(void *user_data, DBusMessageIter* iter);
+static void descriptor_get_flags(void *user_data, DBusMessageIter* iter);
+static void descriptor_get_value(void *user_data, DBusMessageIter* iter);
+
+static void descriptor_read_value(descriptor_t *descriptor, DBusMessageIter* iter);
+
+DBusObjectPathVTable descriptor_dbus_callbacks = {
+  .unregister_function = descriptor_handle_unregister_device,
+  .message_function = descriptor_handle_dbus_message,
+};
+
+static dbus_property_t descriptor_properties[] =
```

Comment:
Same as previous comment about removing the comments or updating them as a block. Currently they dont provide meaningful information